bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/connection/Clip2Bridge.java

```diff
@@ -396,7 +396,7 @@ public void onFailure(@Nullable Session session, @Nullable Throwable failure) {
         @Override
         public void onGoAway(@Nullable Session session, @Nullable GoAwayFrame frame) {
             Objects.requireNonNull(session);
-            if (http2Session == session) {
+            if (session.equals(http2Session)) {
```

Comment:
Both the method 'session' argument and the class 'http2Session' field are @Nullable => therefore the new code could result in an NPE.

Reply:
Oops. I see that line 398 above does actually do the null check..

Reply:
Yes, I think in this file it is ok. In other files without the null check before, I have used ```Objects.equals()```, which is null safe.

Reply:
> I see that line 398 above does actually do the null check..

Yes, and throws NPE if `session` is null. 😉 So this is probably to prevent a SAT warning when it's expected not to be null.

Reply:
I think in reality the passed argument should never be null. The @Nullable annotation is probably an artifact of the crossover between annotated code within OH and non annotated code in the imported library.

Reply:
So .. LGTM ..